app/src/main/java/com/asfoundation/wallet/advertise/AppCoinsAdvertisingBinder.kt

```diff
@@ -22,10 +30,26 @@ internal class AppCoinsAdvertisingBinder(private val packageManager: PackageMana
     val uid = Binder.getCallingUid()
     val pkg = packageManager.getNameForUid(uid)
     val pkgInfo = packageManager.getPackageInfo(pkg, 0)
-    return campaignInteract.getCampaign(pkg, pkgInfo.versionCode).map { mapCampaignDetails(it) }
+    return campaignInteract.getCampaign(pkg, pkgInfo.versionCode)
+        .doOnSuccess { handleNotificationBuild(it) }
+        .map { mapCampaignDetails(it) }
         .blockingGet()
   }
 
+  private fun handleNotificationBuild(campaign: CampaignDetails) {
+    if (campaign.limitReached()) {
```

Comment:
Shouldn't we do this logic in the chain that call this method and the method being used only for the construction of the notification?

Reply:
I don’t agree, since that way you need to call the campaign.limitReached() twice (once to build the leading zeros and another to know if we build or not the notification. But we can talk about this